src/twisted/web/test/test_flatten.py

```diff
@@ -563,3 +569,17 @@ def render(self, request):
         # The original exception is unmodified and will be logged separately if
         # unhandled.
         self.failureResultOf(failing, RuntimeError)
+
+    def test_cancel(self):
```

Comment:
Great to have this test. Thank you for adding it.

I now really want to add some more tests for different cases (i.e. ensuring that the underlying deferred was in fact cancelled, execution order of custom cancelers, etc) but that's a problem for another day :).